member.php

```diff
@@ -1599,8 +1600,35 @@
 			$password_length = 8;
 		}
 
-		$password = random_str($password_length);
-		$logindetails = update_password($user['uid'], md5($password), $user['salt']);
+		// Set up user handler.
+		require_once 'inc/datahandlers/user.php';
+		$userhandler = new UserDataHandler('update');
+
+		$userhandler->set_data(array(
+			'uid'		=> $user['uid'],
+			'username'	=> $user['username'],
+			'email'		=> $user['email'],
+			'password'	=> random_str($password_length)
+		));
+
+		if(!$userhandler->verify_password())
+		{
+			$errors = $userhandler->get_friendly_errors();
+			if(!empty($errors[0]))
+			{
+				error($errors[0]);
+			}
+
+			error($lang->error_unknownerror);
+		}

```

Comment:
Technically, we'd shave off like a millisecond by not verifying it, but from a logical standpoint, I personally like the idea of keeping the check to avoid an odd edge case.